contracts/MarginCalculator.sol

```diff
@@ -35,24 +35,76 @@ contract MarginCalculator {
      * @dev For call return = Max (0, ETH Price - oToken.strike)
      * @dev For put return Max(0, oToken.strike - ETH Price)
      * @param _otoken otoken address
-     * @return the cash value of an expired otoken
+     * @return the cash value of an expired otoken, denomincated in strike asset. scaled by 1e18
      */
-    function getExpiredCashValue(address _otoken) public view returns (uint256) {
+    function _getExpiredCashValue(address _otoken) internal view returns (uint256) {
         require(_otoken != address(0), "MarginCalculator: Invalid token address.");
         OtokenInterface otoken = OtokenInterface(_otoken);
         require(now > otoken.expiryTimestamp(), "MarginCalculator: Otoken not expired yet.");
 
+        // strike price is denominated in strike asset.
         uint256 strikePrice = otoken.strikePrice();
-        (uint256 underlyingPrice, bool isFinalized) = _getUnderlyingPrice(_otoken);
-        require(isFinalized, "MarginCalculator: Oracle price not finalized yet.");
+
+        // divide price of underlying by price of strike,
+        // to get the real price of underlying denominated in strike at expiry
+        (uint256 underlyingPrice, bool isUnderlyingFinalized) = _getAssetPrice(
+            otoken.underlyingAsset(),
+            otoken.expiryTimestamp()
+        );
+        (uint256 strikeAssetPrice, bool isStrikeFinalized) = _getAssetPrice(
+            otoken.strikeAsset(),
+            otoken.expiryTimestamp()
+        );
```

Comment:
so for usdc are we returning 1e18? What happens if the asset doesn't have an oracle?

Reply:
Also if strike == underlying, does this run into the rounding error you were talking about earlier?

Reply:
yeah, for usdc we will need strikeAssetPrice to be 1e18.
We will need 3 oracles for expiry: strike / underlying and collateral. For the most cases, strike or underlying will = collateral, unless we want to do farming option (cETH as collateral...etc)

Reply:
Also re rounding problem, it seems to only happen with `mul`, because all the old test cases passed. I'll need to investigate more into that.